packages/loader/container-loader/src/container.ts

```diff
@@ -592,6 +593,13 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                     [DriverHeader.createNew]: {},
                 };
             }
+            // Add container package name to the request url.
+            const codeDetails = this.context.codeDetails;
+            const packageName = isFluidPackage(codeDetails.package) ? codeDetails.package.name : codeDetails.package;
+            const [mainUrl, queryString] = request.url.split("?");
+            const searchParams = new URLSearchParams(queryString);
+            searchParams.append("containerPackageName", packageName);
+            request.url = `${mainUrl}?${searchParams.toString()}`;
```

Comment:
Why not add it as a header?

Reply:
In @abrahamrq PR: https://github.com/microsoft/FluidFramework/pull/3944 who added this logic in resolver, I think we agreed on having it in URL because we were adding it url in getAbsoluteURL too. I don't think it matters that much until resolver is resolving it properly.

Reply:
ah yeah, I find it easier at least at this point when we're just passing data around to use headers directly instead of some encoding that both sides have to know about. But I'm sure either one works, it just makes it feel like we're not using headers to their full extent :P

Reply:
Jatin is correct, we ended up using the getAbsoluteUrl and that's the reason we used a query param